src/gg.regression.unity.bots/Runtime/Scripts/RGOverlay/RGSegmentEntry.cs

```diff
@@ -0,0 +1,152 @@
+        }
+        
+        if (nameComponent != null)
+        {
+            nameComponent.text = segmentName;
+        }
+        else
+        {
+            Debug.LogError($"RGSegmentEntry is missing its nameComponent");            
+        }
+
+        if (descriptionComponent != null)
+        {
+            descriptionComponent.text = description;
+        }
+        else
+        {
+            Debug.LogError($"RGSegmentEntry is missing its descriptionComponent");
+        }
+
+        if (playButton != null)
+        {
+            playButton.onClick.AddListener(OnPlay);
+        }
+        else
+        {
+            Debug.LogError($"RGSegmentEntry is missing its playButton");
+        }
+
+        // create a tooltip containing the segment's description, but only if the description is populated
+        var tooltip = GetComponentInChildren<RGTooltip>();
+        if (tooltip != null && !string.IsNullOrEmpty(description))
+        {
+            tooltip.content = description;
+        }
+        else
+        {
+            tooltip.SetEnabled(false);
+        }
+
+        // show an icon indicating that this segment is a SegmentList (if needed)
+        if (type == BotSequenceEntryType.SegmentList)
+        {
+            if (segmentListIndicatorComponent != null)
+            {
+                segmentListIndicatorComponent.SetActive(true);
+            }
+            else
+            {
+                Debug.LogError($"RGSegmentEntry is missing its segmentListIndicatorComponent");
+            }
+        }
+    }
+
+    /**
+     * <summary>
+     * When the play button is clicked, start running the Segment and close the RGOverlay
+     * </summary>
+     */
+    private void OnPlay()
+    {
+        var botManager = RGBotManager.GetInstance();
+        if (botManager == null)
+        {
+            Debug.LogError("RGSegmentEntry cannot find the RGBotManager in its OnPlay function");
+            return;
+        }
+        
+        var toolbarManager = FindObjectOfType<ReplayToolbarManager>();
+        if (toolbarManager == null)
+        {
+            Debug.LogError("RGSegmentEntry cannot find the ReplayToolbarManager in its OnPlay function");
+            return;
+        }
+        toolbarManager.selectedReplayFilePath = null;
+        
+        // set the recording toolbar's state for playing this segment
+        botManager.OnBeginPlaying();
+
+        // create the list of segments to play (in our case there will only be 1)
+        var segmentList = new List<BotSegmentList>
+        {
+            BotSequence.CreateBotSegmentListForPath(path, out var sessId)
+        };
+        var sessionId = sessId ?? Guid.NewGuid().ToString();
+        
+        var playbackController = FindObjectOfType<BotSegmentsPlaybackController>();
+        if (playbackController == null)
+        {
+            Debug.LogError("RGSegmentEntry cannot find the BotSegmentsPlaybackController in its OnPlay function");
+            return;

```

Comment:
This function paraphrases how the `BotSequence` class plays a Sequence. I have this feeling that this logic would be good somewhere else.

I'm thinking that playing a Segment should be a static method in the `BotSegment` class. Thoughts and feelings?

Reply:
Re:  // play the segment

Botsequence has this same logic flow in its `Play` method.

I would NOT suggest that you create a sequence for the segments just to re-use this. 

 **Given that playing a bot segment/list directly is only exposed from this UI, I feel like the code is perfect right where it is.**

If you are insistent on moving it somewhere else, it would go in BotSegmentList in a new `Play` method, to match the pattern on `BotSequence`.

Reply:
Your logic is sound, the code will stay put